driver-core/src/main/com/mongodb/internal/connection/AbstractProtocolExecutor.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.internal.connection;
+
+import com.mongodb.MongoOperationTimeoutException;
+import com.mongodb.MongoSocketException;
+import com.mongodb.internal.session.SessionContext;
+
+/**
+ * <p>This class is not part of the public API and may be removed or changed at any time</p>
+ */
+public abstract class AbstractProtocolExecutor implements ProtocolExecutor {
+
+    protected boolean isMongoSocketException(final Throwable e) {
+        return e instanceof MongoSocketException;
+    }
+
+    protected boolean isOperationTimeoutFromSocketException(final Throwable e) {
+        return e instanceof MongoOperationTimeoutException && e.getCause() instanceof MongoSocketException;
```

Comment:
Can `e.getCause()` return null?

Reply:
Yes, it can return null in cases when `MongoOperationTimeoutException` was not caused by any [blocking operation ](https://github.com/mongodb/specifications/blob/master/source/client-side-operations-timeout/client-side-operations-timeout.md#blocking-sections-for-operation-execution).

However, `instanceOf `will return `false` in this case.